event/listener.php

```diff
@@ -66,6 +66,10 @@ static public function getSubscribedEvents()
 	public function load_google_analytics()
 	{
 		$this->template->assign_var('GOOGLEANALYTICS_ID', $this->config['googleanalytics_id']);
+		if ($this->config['googleanalytics_track_user_id'] && $this->user->data['is_registered'])
+		{
+			$this->template->assign_var('GOOGLEANALYTICS_USER_ID', "ga('set', 'userId', {$this->user->data['user_id']});");
+		}
```

Comment:
This shouldn't call assign_var twice. It should instead use assign_vars for both these and set a conditional. And the template code should go in the template. Something more like:

```
$this->template->assign_vars(array(
	'GOOGLEANALYTICS_ID'       => $this->config['googleanalytics_id'],
	'GOOGLEANALYTICS_USER_ID'  => $this->user->data['user_id'],
	'S_GOOGLEANALYTICS_USER_ID => ($this->config['googleanalytics_track_user_id'] && $this->user->data['is_registered']),
));


<!-- IF S_GOOGLEANALYTICS_USER_ID -->
	ga('set', 'userId', {GOOGLEANALYTICS_USER_ID});
<!-- ENDIF -->
```